src/couch_replicator.erl

```diff
@@ -305,3 +296,86 @@ doc_from_db(RepDb, DocId, UserCtx) ->
          {not_found, _Reason} ->
             {error, not_found}
     end.
+
+
+-spec check_authorization(rep_id(), #user_ctx{}) ->
+    authorized | unauthorized | not_found.
+check_authorization(RepId, #user_ctx{name = Name, roles = Roles}) ->
+    case couch_replicator_scheduler:rep_state(RepId) of
```

Comment:
This seems like a race condition.

1) `couch_replicator_scheduler:rep_state/1` is called and returns an `authorized` replication
2) the replication is removed and a new replication is inserted, which the user is not authorized to see
3) the new replication is erroneously cancelled/updated.

Right?